### PR TITLE
Fix search for multiple resource_ids

### PR DESCRIPTION
**Pull request by m-bucher**
@adamruzicka something like that?

see also https://github.com/Katello/katello/pull/8562#discussion_r379355124

### CODE REVIEW

File: app/models/foreman_tasks/task/search.rb [hunk 1]
```diff
@@ -4,7 +4,8 @@ module Search
       def search_by_generic_resource(key, operator, value)
         key = 'resource_type' if key.blank?
         key_name = connection.quote_column_name(key.sub(/^.*\./, ''))
-        condition = sanitize_sql_for_conditions(["foreman_tasks_locks.#{key_name} #{operator} ?", value])
+        value = value.split(',') unless operator.index(/IN/i).nil?
```
Comment (adamruzicka) [inline app/models/foreman_tasks/task/search.rb]: I'd just switch from `unless it does not match` to `if it matches`

**Comment by m-bucher** [comment]
While working on a unit-test for this, I saw that this will return duplicates of the task for every Lock that is found.
So I think we have to add a `distinct` to the search.
However, I do not know how to do this for scoped_search.
@adamruzicka any idea?

**Comment by adamruzicka** [comment]
This cannot be done in scoped search directly. I'd probably add something like

```ruby
scope :for_distinct_search,
      (->(search) { search_for(search).select("DISTINCTI foreman_tasks_tasks.*") })
```

Into `app/models/foreman_tasks/task.rb` and then use `ForemanTasks::Task.for_distinct_search` instead of `ForemanTasks::Task.search_for` when only dinstinct record are needed

**Comment by m-bucher** [comment]
I see, should I add something like that to this PR or should we leave it to those using this search?
I will add a `.distinct` to my test for now.

**Comment by adamruzicka** [comment]
Just add a comment above the search definition that it may return duplicates

**Comment by adamruzicka** [comment]
[test foreman-tasks]

**Comment by adamruzicka** [comment]
Thank you @m-bucher and congratulations to first merged PR into foreman-tasks!

**Comment by jturel** [comment]
very nice!

**Pull request merged**